provision/kubernetes/deploy.go

```diff
@@ -433,17 +436,88 @@ func validateHC(hc *provTypes.TsuruYamlHealthcheck, client *ClusterClient) error
 		if hc.TimeoutSeconds >= hc.IntervalSeconds {
 			hc.IntervalSeconds = hc.TimeoutSeconds + 1
 		}
+
+		if err := ensureBackendConfig(ctx, client, app, hc); err != nil {
+			return err
+		}
+
 	}
 
 	return nil
 }
 
-func probesFromHC(hc *provTypes.TsuruYamlHealthcheck, client *ClusterClient, port int) (hcResult, error) {
+func int64PointerFromInt(v int) *int64 {
+	p := new(int64)
+	*p = int64(v)
+	return p
+}
+
+func ensureBackendConfig(ctx context.Context, client *ClusterClient, a provision.App, hc *provTypes.TsuruYamlHealthcheck) error {
+	exists, err := backendConfigCRDExists(ctx, client)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return nil
+	}
+
+	backendConfigName := fmt.Sprintf("%s-backend-config", a.GetName())
+	cli, err := BackendConfigClientForConfig(client.RestConfig())
+	if err != nil {
+		return err
+	}
+	ns, err := client.AppNamespace(ctx, a)
+	if err != nil {
+		return err
+	}
+
+	intervalSec := int64PointerFromInt(hc.IntervalSeconds)
+	timeoutSec := int64PointerFromInt(hc.TimeoutSeconds)
+	healthyThreshold := int64PointerFromInt(hc.AllowedFailures)
+
+	backendConfig := &backendconfigv1.BackendConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      backendConfigName,
+			Namespace: ns,
+		},
+		Spec: backendconfigv1.BackendConfigSpec{
+			HealthCheck: &backendconfigv1.HealthCheckConfig{
+				CheckIntervalSec:   intervalSec,
+				TimeoutSec:         timeoutSec,
+				HealthyThreshold:   healthyThreshold,
```

Comment:
It's safer to hardcode this as `1` for now since we don't have a specific flag for setting it on regular probes and the default value there is one.